data_steward/cdr_cleaner/cleaning_rules/remove_ehr_data_past_deactivation_date.py

```diff
@@ -11,69 +11,78 @@
-def remove_ehr_data_queries(project_id, ticket_number, pids_project_id,
-                            pids_dataset_id, tablename):
+def remove_ehr_data_queries(client, project_id, dataset_id, sandbox_dataset_id,
+                            fq_deact_table):
     """
     Creates sandboxes and drops all EHR data found for deactivated participants after
     their deactivation date
 
+    :param client: BQ client
     :param project_id: BQ name of the project
-    :param ticket_number: Jira ticket number to identify and title sandbox tables
-    :param pids_project_id: deactivated participants PIDs table in BQ's project_id
-    :param pids_dataset_id: deactivated participants PIDs table in BQ's dataset_id
-    :param tablename: The name of the table to house the deactivated participant data
+    :param dataset_id: Identifies the dataset to retract deactivated participants from
+    :param sandbox_dataset_id: Identifies the sandbox dataset to store records for dataset_id
+    :param fq_deact_table: fully qualified deactivated participants PIDs table in 'project.dataset.table' format
+    :returns queries: List of query dictionaries
     """
-
-    ehr_union_dataset = bq_utils.get_unioned_dataset_id()
-
     # gets the deactivated participant dataset to ensure it's up-to-date
+    pids_project_id, pids_dataset_id, table_name = fq_deact_table.split('.')
     df = psr.get_deactivated_participants(pids_project_id, pids_dataset_id,
-                                          tablename,
+                                          table_name,
                                           DEACTIVATED_PARTICIPANTS_COLUMNS)
     # To store dataframe in a BQ dataset table
-    destination_table = pids_dataset_id + '.' + tablename
+    destination_table = pids_dataset_id + '.' + table_name
     psr.store_participant_data(df, project_id, destination_table)
-    # creates sandbox and truncate queries to run for deactivated participant data drops
-    queries = rdp.create_queries(
-        project_id,
-        ticket_number=ticket_number,
-        # the deactivated participants table is stored in the same project
-        # as the data being retracted
-        pids_project_id=project_id,
-        pids_dataset_id=pids_dataset_id,
-        pids_table=tablename,
-        datasets=[ehr_union_dataset])
 
+    deact_table_ref = gbq.TableReference.from_string(f"{fq_deact_table}")
+    LOGGER.info(f"Retracting deactivated participants from '{dataset_id}'")
+    LOGGER.info(
+        f"Using sandbox dataset '{sandbox_dataset_id}' for '{dataset_id}'")
+    # creates sandbox and truncate queries to run for deactivated participant data drops
+    queries = rdp.generate_queries(client, project_id, dataset_id,
+                                   sandbox_dataset_id, deact_table_ref)
     return queries
 
 
 if __name__ == '__main__':
-    ARGS = rdp.parse_args()
-    pipeline_logging.configure(level=logging.DEBUG,
-                               add_console_handler=ARGS.console_log)
```

Comment:
I wanted to make sure that any data being deleted via deactivation is being monitored, but since this is a cleaning rule it makes sense to ignore it. I'll revert it.